engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/task/ManualTaskProcessor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.bpmn.task;
+
+import io.camunda.zeebe.engine.processing.bpmn.BpmnElementContext;
+import io.camunda.zeebe.engine.processing.bpmn.BpmnElementProcessor;
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnBehaviors;
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnIncidentBehavior;
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnStateTransitionBehavior;
+import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableActivity;
+
+public class ManualTaskProcessor implements BpmnElementProcessor<ExecutableActivity> {
+
+  private final BpmnStateTransitionBehavior stateTransitionBehavior;
+  private final BpmnIncidentBehavior incidentBehavior;
+
+  public ManualTaskProcessor(final BpmnBehaviors bpmnBehaviors) {
+    stateTransitionBehavior = bpmnBehaviors.stateTransitionBehavior();
+    incidentBehavior = bpmnBehaviors.incidentBehavior();
+  }
+
+  @Override
+  public Class<ExecutableActivity> getType() {
+    return ExecutableActivity.class;
+  }
+
+  @Override
+  public void onActivate(final ExecutableActivity element, final BpmnElementContext context) {
+    stateTransitionBehavior.transitionToActivated(context);
+    stateTransitionBehavior.completeElement(context);
```

Comment:
:wrench: We should update the context after a transition to a new state.

The transition to a new state updates the context. We should use the new context in the following steps. Some validations are based on this given context (not here but it would be a possible to verify that the element is activated before completing it). 

```suggestion
  public void onActivate(final ExecutableActivity element, final BpmnElementContext activating) {
    final var activated = stateTransitionBehavior.transitionToActivated(activating);
    stateTransitionBehavior.completeElement(activated);
```